huaweicloud/services/er/resource_huaweicloud_er_static_route.go

```diff
@@ -0,0 +1,219 @@
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The type of the static route.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The current status of the static route.`,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The creation time of the static route.`,
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The latest update time of the static route.`,
+			},
+		},
+	}
+}
+
+func buildStaticRouteCreateOpts(d *schema.ResourceData) routes.CreateOpts {
+	return routes.CreateOpts{
+		Destination:  d.Get("destination").(string),
+		AttachmentId: d.Get("attachment_id").(string),
+		IsBlackHole:  utils.Bool(d.Get("is_blackhole").(bool)),
+	}
+}
+
+func resourceStaticRouteCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.ErV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating ER v3 client: %s", err)
+	}
+
+	var (
+		routeTableId = d.Get("route_table_id").(string)
+		opts         = buildStaticRouteCreateOpts(d)
+	)
+	resp, err := routes.Create(client, routeTableId, opts)
+	if err != nil {
+		return diag.Errorf("error creating static route: %s", err)
+	}
+	d.SetId(resp.ID)
+
+	return resourceStaticRouteRead(ctx, d, meta)
+}
+
+func resourceStaticRouteRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ErV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ER v3 client: %s", err)
+	}
+
+	var (
+		routeTableId  = d.Get("route_table_id").(string)
+		staticRouteId = d.Id()
+	)
+
+	resp, err := routes.Get(client, routeTableId, staticRouteId)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "ER static route")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("destination", resp.Destination),
+		d.Set("is_blackhole", resp.IsBlackHole),
+		// Attributes
+		d.Set("type", resp.Type),
+		d.Set("status", resp.Status),
+		d.Set("created_at", resp.CreatedAt),
+		d.Set("updated_at", resp.UpdatedAt),
+		d.Set("attachment_id", nil), // Override 'attachment_id' while static route is black hole route.
+	)
+	// If the static route is not a black hole route, set related VPC attachment ID.
+	if len(resp.Attachments) > 0 && resp.Attachments[0].AttachmentId != "" {
+		mErr = multierror.Append(mErr, d.Set("attachment_id", resp.Attachments[0].AttachmentId))
+	}
+
+	if mErr.ErrorOrNil() != nil {
+		return diag.Errorf("error saving static route (%s) fields: %s", staticRouteId, mErr)
+	}
+	return nil
+}
+
+func buildStaticRouteUpdateOpts(d *schema.ResourceData) routes.UpdateOpts {
+	var opts = routes.UpdateOpts{}
+
+	if d.HasChange("is_blackhole") && d.Get("is_blackhole").(bool) {
```

Comment:
It is better to add a note for this judgment.

Reply:
emmm...
It seem this logic is unnecessary, it is old school for local branch.